src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WorkspaceSemanticTokensRefreshNotifier.cs

```diff
@@ -30,11 +30,13 @@ internal class WorkspaceSemanticTokensRefreshNotifier : IWorkspaceSemanticTokens
         IClientConnection clientConnection,
         RazorLSPOptionsMonitor optionsMonitor)
     {
-        _clientCapabilitiesService = clientCapabilitiesService;
         _clientConnection = clientConnection;
 
         _disposeCancellationSource = new();
 
+        var capabilities = clientCapabilitiesService.ClientCapabilities;
```

Comment:
Sorry, I should have been clearer in my previous comment. I expect this won't work in the constructor, as we can't call `ClientCapabilites` until the `initialize` LSP message has been received, so we still need the code in the notify method, just it doesn't need to happen inside the lock.

Unless this class is not constructed eagerly, or in DI, in which case perhaps the point is moot.

Reply:
Oh, I see. Thank!